h/static/scripts/streamer.js

```diff
@@ -68,7 +68,9 @@ function Streamer($rootScope, annotationMapper, annotationUI, features, groups,
     case 'delete':
       annotations.forEach(function (ann) {
         delete pendingUpdates[ann.id];
-        pendingDeletions[ann.id] = true;
+        if (annotationUI.annotationExists(ann.id)) {
```

Comment:
Hmm. Seems a little odd to me that we're explicitly checking by ID for deletes, but not for updates.